created a shell script to restart download upon failure

mvish7 opened the pull request:
Hi
I'm downloading the dataset using the contrib/99991/download.py script. The downloading got repeatedly interrupted due to an error of "remote not responding". 

Hence I created a simple shell script to restart download upon failure.

mvish7 committed:
created a shell script to restart download upon failure

mikeroberts3000 commented:
Hi, thanks for submitting this PR. When you get these `remote not responding` errors, what happens to the file that was being downloaded at the time of the error?

mvish7 committed:
removed shell script

mvish7 committed:
adapted the script to get around connection issues

mvish7 committed:
added new requirements for download script

mvish7 committed:
added logic to retry on connection failure errors

mvish7 commented:
Hi
Sorry for a super late reply and thanks for the hint. 

I inspected the download.py script and currently if the connection ends abruptly then the files will be partially downloaded.  I introduced a way to catch these errors + to remove the partially downloaded file and download the same file. I have added two functions for this

- download_curr_file: This function together with added logic in `download_files` function handles removing partially downloaded file and downloading the same file until it is successfully downloaded.
- init_webfile: This function along with the added logic in `download_files` function catches the "bad zip file" errors.

With these changes I could successfully all the `depth_meters.hdf5` and `color.hdf5 files from `ai_043_004` onwards. Hence I have removed the  `download.sh` script which was responsible for restarting the `download.py` after failure.

mikeroberts3000 commented:
I've decided not to merge this PR because: (1) the added functionality targets a narrow and potentially user-specific use case; (2) I'm not sure if the proposed code changes are correct; and (3) there is a stylistic mismatch between the code in the PR and the rest of the code in this repository.
